File: yh_tutorial/yh_tutorial_5/src/yh_client_5.cpp
#include "ros/ros.h"
#include "yh_tutorial_5/yh_srv_5.h"
#include <cstdlib>

int main(int argc, char ** argv)
{
    ros::init(argc, argv, "yh_client_5");

    if (argc != 3)
    {
        ROS_INFO("wrong command");
        return 1;
    }

    ros::NodeHandle nh;

    ros::ServiceClient service_client = nh.serviceClient<yh_tutorial_5::yh_srv_5>("yh_service_5");

    yh_tutorial_5::yh_srv_5 srv;

    srv.request.a = atoll(argv[1]);
    srv.request.b = atoll(argv[2]);

    if (service_client.call(srv))
    {
        ROS_INFO("%ld", srv.response.result);
    }
    else
    {
        ROS_ERROR("Failed");
        return 1;
    }

    return 0;
}

File: yh_tutorial/yh_tutorial_5/scripts/yh_client_5.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_5.srv import yh_srv_5
import sys

def yh_client(a, b):
    rospy.wait_for_service("yh_service_5")
    try:
        service_client = rospy.ServiceProxy("yh_service_5", yh_srv_5)
        res = service_client(a, b)
        return res.result
    except rospy.ServiceException as e:
        rospy.logerr(e)

if __name__ == "__main__":
    rospy.init_node("yh_client_5")
    if len(sys.argv) != 3:
        rospy.loginfo("wrong")
        sys.exit(1)
    
    a = int(sys.argv[1])
    b = int(sys.argv[2])
    res = yh_client(a, b)
    rospy.loginfo("%d", res)